src/dune_rules/lib.mli

```diff
@@ -123,8 +123,16 @@ module DB : sig
 
   val find : t -> Lib_name.t -> lib option Memo.t
   val find_even_when_hidden : t -> Lib_name.t -> lib option Memo.t
+
+  (** [available db lib_name] returns [true] if there is any library available
+      in the database with name [lib_name], regardless if it is enabled or not. *)
   val available : t -> Lib_name.t -> bool Memo.t
 
+  (** [available_library_stanza db ~dir lib_name], similar to [available], but only
+      returns [true] if there is a library available with name [lib_name], and the stanza
+      directory where the library is defined matches [dir]. *)
+  val available_library_stanza : t -> dir:Path.Build.t -> Lib_name.t -> bool Memo.t
```

Comment:
`Lib.DB.get_compile_info` needs to have this `~dir` argument as well. Otherwise we might return the compilation information of the wrong library.

Reply:
I tried to get something done about this in https://github.com/ocaml/dune/pull/9839/commits/28585d63a9dcd23d5ceb1f4fe096faea60fd174a.